### PR TITLE
[ci][python] improved paths in setup.py and small CI refactoring

### PR DESCRIPTION
**Pull request by StrikerRUS**
Made paths in `setup.py` more readable and platform independent.

Less `cd` commands at Appveyor and full paths at Travis.

Unfortunately, we still cannot perform installation of Python-package from other directory due to relative paths in `Manifest.in`.

### CODE REVIEW

File: python-package/setup.py [hunk 1]
```diff
@@ -32,25 +31,31 @@ def find_lib():
...
-        distutils.file_util.copy_file("../windows/LightGBM.sln", "./compile/windows/LightGBM.sln")
-        distutils.file_util.copy_file("../windows/LightGBM.vcxproj", "./compile/windows/LightGBM.vcxproj")
+        if not os.path.exists(os.path.join(CURRENT_DIR, "compile", "windows")):
+            os.makedirs(os.path.join(CURRENT_DIR, "compile", "windows"))
+        distutils.file_util.copy_file(os.path.join(CURRENT_DIR, os.path.pardir, "windows", "LightGBM.sln"),
+                                      os.path.join(CURRENT_DIR, "compile", "windows", "LightGBM.sln"))
+        distutils.file_util.copy_file(os.path.join(CURRENT_DIR, os.path.pardir, "windows", "LightGBM.vcxproj"),
+                                      os.path.join(CURRENT_DIR, "compile", "windows", "LightGBM.vcxproj"))
+        distutils.file_util.copy_file(os.path.join(CURRENT_DIR, os.path.pardir, "windows", "LightGBM.vcxproj.filters"),
+                                      os.path.join(CURRENT_DIR, "compile", "windows", "LightGBM.vcxproj.filters"))
```
Comment (StrikerRUS) [inline python-package/setup.py]: @guolinke Do we need to copy this file too?

Refer to https://github.com/Microsoft/LightGBM/commit/7689a4d0647b07cc3357e28e3eb85ae24d7e919a

Comment (guolinke) [reply python-package/setup.py]: I think `filters` is not needed for build

**Comment by StrikerRUS** [comment]
@guolinke Removed copying of `filters` file and refined paths in nugget creation script.

**Comment by StrikerRUS** [comment]
@guolinke Because you are saying that we don't need `LightGBM.vcxproj.filters`, I've removed it from `MANIFEST.IN` file.

**Pull request merged**